### PR TITLE
Try to fix test many vars

### PR DESCRIPTION
**Pull request by scottwittenburg**

### CODE REVIEW

**Comment by scottwittenburg** [comment]
Ping @pnorbert I branched this off your topic (#470) that was failing on appveyor.  My commit here seems to fix the build errors, so you may just want to copy those two lines, since they belong in a file that won't exist in the repo until your PR is merged.  Or we can just merge this PR.

But lets see if @chuckatkins has some thoughts about this issue before we proceed.

File: thirdparty/GTest/CMakeLists.txt [hunk 1]
```diff
@@ -6,5 +6,5 @@ set(CMAKE_POLICY_DEFAULT_CMP0042 NEW)
 add_subdirectory(googletest EXCLUDE_FROM_ALL)
 
 if(WIN32 AND BUILD_SHARED_LIBS)
-  target_compile_definitions(gtest INTERFACE GTEST_LINKED_AS_SHARED_LIBRARY=1)
+  target_compile_definitions(gtest_main INTERFACE GTEST_LINKED_AS_SHARED_LIBRARY=1)
```
Comment (scottwittenburg) [inline thirdparty/GTest/CMakeLists.txt]: Thanks @chuckatkins this did the trick.  I'm going to rewrite the history of this PR cleanly, then we could even merge and let @pnorbert rebase his topic off master.

**Comment by scottwittenburg** [comment]
Ok I rewrote the history of this into a single commit after the tests had all passed.  But I also rebased this topic (which includes the #470 topic by @pnorbert) of the latest master, which seems to have caused the new build failures on CircleCI (should be failing on AppVeyor too, not sure why that is still yellow).

So I pushed a new topic (#477) with just my commit, which is based on master.  I was thinking I'd close this topic, get #477 merged, and then let @pnorbert rebase off master at that point and fix the issues in his topic that have arisen since master changed.  Thoughts?

Ping @chuckatkins

**Comment by chuckatkins** [comment]
Agreed. #477 is merged.

**Pull request closed without merging**